docs/proposals/ldap-group-sync.md

```diff
@@ -0,0 +1,187 @@
+# Allowing OpenShift Groups to be Synced With Sources on an LDAP Server and Making it Extensible for Other External Sources
+
+___
+Note: in this proposal, if a reference is prefixed with "OpenShift" or is typeset in `monospace`, it refers to an OpenShift object or field. For instance, the following table enumerates what references to OpenShift resources and LDAP entries may look like:
+
+OpenShift Resource | LDAP Entry
+------------------ | ----------
+OpenShift Group    | LDAP group
+OpenShift `Group`  | group
+`Group`            |
+___
+
+Linking OpenShift `Groups` to their representations on LDAP servers allows for administrators to have an easier way to manage `Groups` inside OpenShift. This additional functionality will require no additional fields on the OpenShift `Group` object. One can indicate that an OpenShift `Group` mirrors an external representation by setting the `externalRecordProvider` `Annotation` in the `Group` as well as adding an `externalRecordProvider` `Label` such as `openshift.io/externalRecordProvider/LDAP` to the `Group`. Other information (e.g. last sync time or resource version) will be stored as `Annotations` on the OpenShift `Group`. Logistical information for the LDAP sync will be stored in an `LDAPSyncOptions` that the administrator will need to provide.
+
+### Use Cases
+* I want to sync all groups I've configured in a sync job (e.g. a specific LDAP server configuration) to OpenShift `Groups`
+* I want to sync specific OpenShift `Groups` with their `externalRecordProviders`
+
+### Data Flow
+* Determine list of groups to be synced
+* Find LDAP group members
+* Determine OpenShift `Identities` from LDAP group members
+* Determine OpenShift `Users` from OpenShift `Identities`
+* Determine OpenShift `Group` from LDAP group
```

Comment:
Running `sync` on a group you specify and having it update a different group and *not* the one you asked for - which is implied by non-deterministic mapping from the LDAP response to the OpenShift `Group` - seems bad.